From 1f403bd1d7867e5325450251fe6c31ad958ff3b4 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:10:42 -0400
Subject: [usb] pl2303: add error handling to set_control_lines

Message-id: <1410887463-20674-571-git-send-email-dzickus@redhat.com>
Patchwork-id: 95485
O-Subject: [RHEL7 PATCH 570/591] USB: pl2303: add error handling to set_control_lines
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit a6ec8245bf09fd51a0561ff372a12473b48d269b
Author: Johan Hovold <jhovold@gmail.com>
Date:   Sun Dec 29 19:23:02 2013 +0100

    USB: pl2303: add error handling to set_control_lines

    Add error handling to set_control_lines.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/pl2303.c b/drivers/usb/serial/pl2303.c
index 27756e2..0e7199f 100644
--- a/drivers/usb/serial/pl2303.c
+++ b/drivers/usb/serial/pl2303.c
@@ -269,12 +269,13 @@ static int pl2303_set_control_lines(struct usb_serial_port *port, u8 value)
  struct usb_device *dev = port->serial->dev;
  int retval;
 
+ dev_dbg(&port->dev, "%s - %02x\n", __func__, value);
+
  retval = usb_control_msg(dev, usb_sndctrlpipe(dev, 0),
      SET_CONTROL_REQUEST, SET_CONTROL_REQUEST_TYPE,
      value, 0, NULL, 0, 100);
-
- dev_dbg(&port->dev, "%s - value = %d, retval = %d\n", __func__,
-  value, retval);
+ if (retval)
+  dev_err(&port->dev, "%s - failed: %d\n", __func__, retval);
 
  return retval;
 }
-- 
1.7.1